src/routes/strategy/wallet-dashboard.svelte

```diff
@@ -0,0 +1,110 @@
+<script>
+	import { ethers, utils } from 'ethers';
+	import { InputGroup, InputGroupText, Input, Spinner } from 'sveltestrap';
+	import { formatUSDCBalance } from '$lib/helpers/formatters';
+	import erc20ABI from '../../lib/abi/erc20.json';
+
+	let metamaskConnected = false;
+	let usdcBalance = null;
+	let usdcBalanceFetched = false;
+
+	let walletConnected = false;
+	let connectWalletError = false;
+	let account = '0x000...000';
+	let provider = null;
+	let signerInit = null;
+
+	const usdcAddress = '0x2791bca1f2de4661ed88a30c99a7a9449aa84174';
+
+	async function getAvailableToInvest(web3, signer, selectedAccount) {
+		const usdcContract = new ethers.Contract(usdcAddress, erc20ABI, web3);
+		const balance = await usdcContract.balanceOf(selectedAccount);
+		usdcBalanceFetched = true;
+		return utils.formatEther(balance);
+	}
+
+  async function handleWalletConnection() {
+		if(!walletConnected) {
+			connectWallet();
+		} else {
+			disconnectWallet();
+		}
+	}
```

Comment:
This method is never called. Did you mean to use this instead of `connectWallet` as the button handler in line `93`?

I tried changing to `handleWalletConnection` in the button handler and it doesn't behave quite right. I suspect this is related to the two different flags (`walletConnected` and `metamaskConnected`) being out-of-sync.

Also… does this method need to be `async`? (it never uses `await` keyword, so it shouldn't need to be `async`)

Reply:
@kenkunz thanks for the review. I did some changes based on your review. 

About the design, I agree we need to discuss how it will be. At the moment there is no link to reach this page in the frontend.

About the USDC balance a 0 is because we have to create a better mechanism to check the smart contract in multiple API, in the eth lisbon we used polygon, so we need to think how are we going to implement a blockchain switcher button for Ethereum, polygon, bnb smart chain etc ...

Reply:
yep I changed to handle the connnection on the click event and removed the async.